src/main/java/de/dhbw/mh/slang/craft/parser/CraftedSlangParser.java

```diff
@@ -332,29 +335,83 @@ AstNode signedTerm3( ){
 	/*===========================================================
 	 * exponentiation
 	 *===========================================================*/
-	
+
+	// Initial Rules:
+	// Exponentiation: atomicExpression "**" exponentiation
+	// Exponentiation: eps
+
+	// There is left-recursion, as the atomicExpression circles over many rules again to Exponentiation
+	// This is noticeable in the select set
+
+	// Rules without left recursion
+	// Exponentiation: atomicExpression Exponent
+	// Exponent: "**" exponentiation
+	// Exponent: eps
+
+
 	@Override
 	public AstNode exponentiation( ){
-		// TODO Auto-generated method stub
-		return super.exponentiation( );
+		// Rule: Exponentiation: atomicExpression Exponent
+		// Select Set: LPAREN, IDENTIFIER, NUMERIC_LITERAL, EOF
+
+		if (LPAREN == LEXER.lookahead().TYPE
+				|| IDENTIFIER == LEXER.lookahead().TYPE
+				|| NUMERIC_LITERAL == LEXER.lookahead().TYPE
+				|| EOF == LEXER.lookahead().TYPE) {
+			return exponent(atomicExpression());
+		}
+
+		Set<Token.Type> acceptedTypes = setOf(LPAREN, IDENTIFIER, NUMERIC_LITERAL, EOF);
+		throw this.parsingException(acceptedTypes);
 	}
 	
 	@Override
 	public AstNode exponent( AstNode previous ){
-		// TODO Auto-generated method stub
-		return super.exponent( previous );
+		// This function decides which of the both rules for "exponent" should be applied
+		// This is determined by the select set
+
+
+		switch (LEXER.lookahead().TYPE) {
+
+			// Rule: Exponent: "**" Exponentiation
+			// Select Set: POWER
+			case POWER:
+				return exponent1(previous);
+
+			// Rule: Exponent: eps
+			// Select Set: +,-,*,/ ..
+			case PLUS:
+			case MINUS:
+			case ASTERISK:
+			case DIVIDE:
+			case GREATER:
+			case LESS:
+			case GREATER_EQUAL:
+			case LESS_EQUAL:
+			case EQUAL:
+			case NOT_EQUAL:
+			case LAND:
+			case LOR:
+			case RPAREN:
+			case MODULO:
+			case EOF:
```

Comment:
I don't think that's quite right, but obviously I'm assuming just these tokens in the test 🤔 Looks like I made a mistake here... Please forgive. I'll merge your change anyway so we can talk about it on Tuesday.